packages/core/src/sdk/product/useAllVariantProducts.ts

```diff
@@ -0,0 +1,112 @@
+import { useMemo } from 'react'
+
+import { gql } from '@generated'
+import type {
+  ClientAllVariantProductsQueryQuery,
+  ClientProductQueryQueryVariables,
+} from '@generated/graphql'
+
+import { useQuery } from '../graphql/useQuery'
+import { useSession } from '../session'
+
+const query = gql(`
+  query ClientAllVariantProductsQuery($locator: [IStoreSelectedFacet!]!) {
+      product(locator: $locator) {
+      ...ProductSKUMatrixSidebarFragment_product
+    }
+  }
+`)
```

Comment:
The queries that we use in `faststore/core` can be extendable using the [API Extension feature](https://developers.vtex.com/docs/guides/faststore/api-extensions-extending-queries-using-fragments). This query in this way is not part of the Extendable feature. 

I wonder if we should implement it or not. cc @renatomaurovtex @vini (idk vini's handle hahah, I'll send this question to him directly)

Reply:
Ok, I'm wait this definition

Reply:
Hello, we've discussed this internally and have decided to address this query as an extendable feature in our backlog at another moment.

Reply:
👍